molecule/default/roles/helm/tasks/tests_chart.yml

```diff
@@ -1,321 +1,358 @@
+        namespace: "{{ helm_namespace }}"
+      register: install
+
+    - name: "Assert that {{ chart_test }} chart is installed from {{ source }}"
+      assert:
+        that:
+          - install is changed
+
+    - name: Remove {{ chart_test }} with --purge
+      helm:
+        binary_path: "{{ helm_binary }}"
+        state: absent
+        name: test-0001
+        purge: False
+        namespace: "{{ helm_namespace }}"
+      register: install
+
+    - name: Check if chart is removed
+      assert:
+        that:
+          - install is changed
+
+    - name: Install chart again with same name test-0001
+      helm:
+        binary_path: "{{ helm_binary }}"
+        name: test-0001
+        chart_ref: "{{ chart_source }}"
+        chart_version: "{{ chart_source_version | default(omit) }}"
+        namespace: "{{ helm_namespace }}"
+        replace: True
+      register: install
+
+    - name: "Assert that {{ chart_test }} chart is installed from {{ source }}"
+      assert:
+        that:
+          - install is changed
+
+    - name: Remove {{ chart_test }} (cleanup)
+      helm:
+        binary_path: "{{ helm_binary }}"
+        state: absent
+        name: test-0001
+        namespace: "{{ helm_namespace }}"
+      register: install
+
+    - name: Check if chart is removed
+      assert:
+        that:
+          - install is changed
+
+    - name: "Install {{ chart_test }} from {{ source }} with values_files"
+      helm:
+        binary_path: "{{ helm_binary }}"
+        name: test
+        chart_ref: "{{ chart_source }}"
+        chart_version: "{{ chart_source_version | default(omit) }}"
+        namespace: "{{ helm_namespace }}"
+        values_files:
+          - "{{ role_path }}/files/values.yaml"
+      register: install
+
+    - name: "Assert that {{ chart_test }} chart has var from {{ source }}"
+      assert:
+        that:
+          - install is changed
+          - install.status.status | lower == 'deployed'
+          - install.status.chart == "{{ chart_test }}-{{ chart_test_version }}"
+          - "install.status['values'].revisionHistoryLimit == 0"
+
+    - name: "Install {{ chart_test }} from {{ source }} with values_files (again)"
+      helm:
+        binary_path: "{{ helm_binary }}"
+        name: test
+        chart_ref: "{{ chart_source }}"
+        chart_version: "{{ chart_source_version | default(omit) }}"
+        namespace: "{{ helm_namespace }}"
+        values_files:
+          - "{{ role_path }}/files/values.yaml"
+      register: install
+
+    - name: "Assert the result is consistent"
+      assert:
+        that:
+          - not (install is changed)
+
+    - name: Render templates
+      helm_template:
+        binary_path: "{{ helm_binary }}"
+        chart_ref: "{{ chart_source }}"
+        chart_version: "{{ chart_source_version | default(omit) }}"
+        output_dir: "{{ temp_dir }}"
+        values_files:
+          - "{{ role_path }}/files/values.yaml"
+      register: result
```

Comment:
It would be nice to have some assertions for this task.  is not failed, result.rc value, and result.command should be straightforward, depending on the reproducability and complexity of result.stdout there should be something we can assert about the shape or contents that.

Reply:
Unfortunately, examining stdout here will be difficult. This test suite gets run multiple times with different charts so the output will be different for each one. The same thing goes for examining `result.command` to some extent. I've added a check for as much of the command as we can reliably predict.